### PR TITLE
V3 Online Interpreter

### PR DESCRIPTION
**Pull request by Lyxal**
This gets the website infrastructure ready. It currently uses the v2 styling and v2 vylight. That's okay for now though.

### CODE REVIEW

File: .github/workflows/compile-to-js.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK 17
+      uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'adopt'
+    - name: Scala to JS
+      run: sbt fastOptJS
```
Comment (ysthakur) [inline .github/workflows/compile-to-js.yaml:20]: This should be in the deploy-pages workflow

File: .github/workflows/deploy-pages.yaml [hunk 2]
```diff
@@ -39,6 +39,15 @@ jobs:
...
           distribution: 'adopt'
+      
+      - name: Scala to JS
+        run: sbt fastOptJS
+      - name: Stage changes
+        run: git add .
+      - name: Commit and push
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: "Reroll JS from Scala"
```
Comment (ysthakur) [inline .github/workflows/deploy-pages.yaml:50]: (sorry, no suggestions on drafts or something)
```yaml
- name: Scala to JS
   run: sbt fullOptJS
```

File: .github/workflows/deploy-pages.yaml [hunk 3]
```diff
@@ -3,7 +3,7 @@ name: Generate docs
 on:
   push:
     branches:
-      - scaladoc # todo change this to the "main" v3 branch
+      - v3-online-interpreter # todo change this to the "main" v3 branch
```
Comment (ysthakur) [inline .github/workflows/deploy-pages.yaml:6]:
```suggestion
      - version-3
```
Leaving this suggestion here to apply before merging

**Review by ysthakur — APPROVED** [review]

**Comment by ysthakur** [comment]
I'm going to have to reject this PR because [this](https://vyxal.github.io/Vyxal/#WyIiLCIiLCJseXhhbCIsIiIsIiJd) doesn't work anymore.

**Comment by Lyxal** [comment]
> I'm going to have to reject this PR because [this](https://vyxal.github.io/Vyxal/#WyIiLCIiLCJseXhhbCIsIiIsIiJd) doesn't work anymore.

This has been fixed in commit 1ba12a7

**Review by Steffan153 — APPROVED** [review]
We can rewrite this right?

**Comment by Steffan153** [comment]
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA I GOT RICKROL-
@ysthakur DELETE THAT TERRIBLE LINK

**Comment by Lyxal** [comment]
> We can rewrite this right?

Of course! The entire styling of the site is up to change.

**Comment by ysthakur** [comment]
@Steffan153 In my defense, it wasn't working when I posted it :P

**Pull request merged**